src/screens/transaction-details/components/TransactionDetailsContent.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import { Bleed, Box, Separator } from '@/design-system';
+import { SingleLineTransactionDetailsRow } from '@/screens/transaction-details/components/SingleLineTransactionDetailsRow';
+import { shortenTxHashString } from '@/screens/transaction-details/helpers/shortenTxHashString';
+import { DoubleLineTransactionDetailsRow } from '@/screens/transaction-details/components/DoubleLineTransactionDetailsRow';
+import { TransactionDetailsSymbol } from '@/screens/transaction-details/components/TransactionDetailsSymbol';
+import { getFormattedEthFee } from '@/screens/transaction-details/helpers/getFormattedEthFee';
+
+type Props = {
+  accountAddress: string;
+  txHash?: string;
+  fromCurrentAddress?: boolean;
+  weiFee?: number;
+};
+
+const Divider = () => (
+  <Bleed horizontal="20px">
+    <Separator color="separatorTertiary" />
+  </Bleed>
+);
+
+export const TransactionDetailsContent: React.FC<Props> = ({
+  accountAddress,
+  txHash,
+  fromCurrentAddress,
+  weiFee,
+}) => {
+  const formattedHash = txHash ? shortenTxHashString(txHash) : '';
+  const formattedEthFee = getFormattedEthFee(weiFee ?? 0);
+
+  return (
+    <Box
+      background="surfacePrimary"
+      flexGrow={1}
+      paddingHorizontal="20px"
+      paddingBottom="20px"
+    >
+      {fromCurrentAddress && weiFee && (
```

Comment:
this feels weird to me, I ~think these should be wrapped in their own typed components, `TxDetailsFeeRow`and `TxDetailsHashRow` , not really sure what else will be needed but just a thought. 

my main smell I dont like here is that were checking hash vs fee and then showing content based on those, is this gonna grow to more types or are these interchangeable ?

Reply:
Hmmm, this is a good question. But if they grew to more types and we had specific typed components for each, wouldn't that grow to more code than re-using generic components with different values like we have here atm?